src/components/TodoList.jsx

```diff
@@ -1,66 +1,206 @@
+            ...this.state
+        };
         this.setState(newState);
     }
-
+    // Add task
     addTask() {
-        const { tasks } = this.state;
-
+        const { tasks, sortingOption, lastCreatedId } = this.state;
+        // Set properties of the new task
+        const newTask = {
+            taskName: this.state.input,
+            date: new Date(),
+            id: lastCreatedId + 1,
+            color: "",
+            textDecoration: ""
+        }
+        // Sort task as it added
+        const newTasks = tasks.concat(newTask).sort(filterTasks(sortingOption));
         this.setState({
             input: "",
-            tasks: tasks.concat(this.state.input)
+            tasks: newTasks,
+            lastCreatedId: lastCreatedId + 1
         });
     }
 
     onInputChange(event) {
         const text = event.target.value;
-        this.setState({ ...this.state, input: text });
+        this.setState({ ...this.state, input: text })
+    }
+
+    removeAllTasks() {
+        this.setState(
+            {
+                ...this.state,
+                tasks: []
+            }
+        )
+    }
+    mapTasks(mapArgument) {
+        return this.state.tasks.map(mapArgument)
+    }
+    // Key of task html element 
+    getKeyForTodoElem(todoPoint) {
+        return this.state.tasks.indexOf(todoPoint)
     }
 
     render() {
-        return <div>
+        return <>
             <div>
+                <select
+                    onChange={this.sortTasks}>
+                    {
+                        sortingOptions.map(option =>
+                            <option
+                                key={sortingOptions.indexOf(option)}>
+                                {option}
+                            </option>)
+                    }
+                </select>
                 <button onClick={this.addTask}>Add Task</button>
-                <input type="text" value={this.state.input} onChange={this.onInputChange}></input>
+                <input
+                    type="text"
+                    value={this.state.input}
+                    onChange={this.onInputChange}
+                />
+                <button onClick={this.removeAllTasks}>Remove Tasks</button>
             </div>
-            <ul style={todoStyle}>{
-                this.state.tasks.map(text => <TodoPoint key={text} todoText={text} removeTask={this.removeTask} />)
-            }</ul>
-        </div>
-
+            <ul id="todoPoints"
+                style={todoStyle} >{
+                    this.mapTasks(todoPoint => <TodoPoint
+                        key={this.getKeyForTodoElem(todoPoint)}
+                        todoText={todoPoint.taskName}
+                        todoId={todoPoint.id}
+                        bIsEdited={todoPoint.id === this.state.taskInEditId}
+                        removeTask={this.removeTask}
+                        todoDate={formatDate(todoPoint)}
+                        taskInEditText={this.state.taskInEditText}
```

Comment:
This prop is not used anywhere in the code. Please remove props you are not using, this code has become very bloated.

Reply:
Whoops, this one IS used. The one not used is `taskInEditID`.

Reply:
Altho no, looks like they are both not used :)